PurchasesCoreSwift/Public/PurchaserInfo.swift

```diff
@@ -0,0 +1,315 @@
+        self.managementURL = subscriberData.managementURL
+        self.originalApplicationVersion = subscriberData.originalApplicationVersion
+
+        self.nonSubscriptionTransactions = subscriberData.nonSubscriptionTransactions
+
+        self.entitlements = EntitlementInfos(entitlementsData: subscriberData.entitlementsData,
+                                             purchasesData: subscriberData.allTransactionsByProductId,
+                                             dateFormatter: Self.dateFormatter,
+                                             requestDate: requestDate)
+
+        self.expirationDatesByProductId = subscriberData.expirationDatesByProductId
+        self.purchaseDatesByProductId = subscriberData.purchaseDatesByProductId
+    }
+
+    // TODO after migration make this internal
+    @objc public static let currentSchemaVersion = "2"
+
+    /**
+     Get the expiration date for a given product identifier. You should use Entitlements though!
+    
+     @param productIdentifier Product identifier for product
+    
+     @return The expiration date for `productIdentifier`, `nil` if product never purchased
+     */
+    @objc public func expirationDate(forProductIdentifier productIdentifier: String) -> Date? {
+        return expirationDatesByProductId[productIdentifier] ?? nil
+    }
+
+     /**
+     Get the latest purchase or renewal date for a given product identifier. You should use Entitlements though!
+    
+     @param `productIdentifier` Product identifier for subscription product
+    
+     @return The purchase date for `productIdentifier`, `nil` if product never purchased
+     */
+    @objc public func purchaseDate(forProductIdentifier productIdentifier: String) -> Date? {
+        return purchaseDatesByProductId[productIdentifier] ?? nil
+    }
+
+    /**
+     Get the expiration date for a given entitlement.
+    
+     @param` entitlementIdentifier` The id of the entitlement.
+    
+     @return The expiration date for the passed in `entitlement`, can be `nil`
+     */
+    @objc public func expirationDate(forEntitlement entitlementIdentifier: String) -> Date? {
+        return entitlements[entitlementIdentifier]?.expirationDate
+    }
+
+    /**
+     Get the latest purchase or renewal date for a given entitlement identifier.
+    
+     @param `entitlementIdentifier` Entitlement identifier for entitlement
+    
+     @return The purchase date for `entitlement`, `nil` if product never purchased
+     */
+    @objc public func purchaseDate(forEntitlement entitlementIdentifier: String) -> Date? {
+        return entitlements[entitlementIdentifier]?.latestPurchaseDate
+    }
+
+    // TODO after migration make this internal and remove objc rename
+    @objc(JSONObject) public func jsonObject() -> [String: Any] {
+        return originalData.merging(
+            ["schema_version": PurchaserInfo.currentSchemaVersion],
+            uniquingKeysWith: { (current, _) in current })
+    }
```

Comment:
perhaps we can keep all of these will-be-internal _after_ the public ones? that way we're following the public first, internal second, private last structure